domain/blockdevice/service/service.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/collections/set"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/core/watcher/eventsource"
+	"github.com/juju/juju/domain/blockdevice"
+	"github.com/juju/juju/domain/blockdevice/state"
+)
+
+// State defines an interface for interacting with the underlying state.
+type State interface {
+	BlockDevices(ctx context.Context, machine string) ([]blockdevice.BlockDevice, error)
+	GetMachineInfo(ctx context.Context, machine string) (string, int, error)
```

Comment:
I think this int should be a stronger enum type for the lifecycle of the machine.

Reply:
yup, was the todo but fixed now